kanren/cons.py

```diff
@@ -0,0 +1,246 @@
+from functools import reduce
+
+from itertools import chain
+
+from collections import OrderedDict
+
+try:
+    from operator import length_hint
+except ImportError:
+    def length_hint(obj, default=0):
```

Comment:
Do you need Python 2 for your use case? Unless @mrocklin objects, I'd support dropping Python 2 support rather than adding compatibility code. Python 2.7 has only 9 months left, and lots of notable libraries are dropping support for it (ipython, django, etc.)

Reply:
I would love to drop Python 2 support, but it's in the CI process, so I added it.  If you want, I can drop Python 2 from CI in this PR as well.

Reply:
Sounds good to me. Maybe also drop 3.4, since it has an end of life in 11 days. https://devguide.python.org/#status-of-python-branches

Reply:
Done.